File: Level2/221102/Java/RemovePairTest.java
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class RemovePairTest {
  @Test
  void solution() {
    RemovePair removePair = new RemovePair();

    assertEquals(1, removePair.solution("baabaa"));
    assertEquals(0, removePair.solution("cdcd"));
  }
}

File: Programmers/Level2/221103/JavaScript/RemovePair.test.js
import { remove, solution, toArray } from './RemovePair';

test('wordsToArray', () => {
  expect(toArray('baabaa')).toStrictEqual(['b', 'a', 'a', 'b', 'a', 'a']);
});

test('removeWord', () => {
  expect(remove(['b', 'a', 'a', 'b', 'a', 'a'])).toStrictEqual([]);
  expect(remove(['c', 'd', 'c', 'd'])).toStrictEqual(['c', 'd', 'c', 'd']);
});

test('solution', () => {
  expect(solution('baabaa')).toBe(1);
  expect(solution('cdcd')).toBe(0);
});
